port/zephyr/golioth_sys_zephyr.c

```diff
@@ -193,6 +194,10 @@ void golioth_sys_thread_destroy(golioth_sys_thread_t gthread) {
  * Misc
  *------------------------------------------------*/
 
-void golioth_sys_client_connected(void* client) {}
+void golioth_sys_client_connected(void* client) {
+    log_backend_golioth_enable(client);
+}
 
-void golioth_sys_client_disconnected(void* client) {}
+void golioth_sys_client_disconnected(void* client) {
+    log_backend_golioth_disable(client);
+}
```

Comment:
Since those functions are implemented by platform-dependent code (Linux, Zephyr, ...) maybe we should put a `_platform_` in the function name?

Reply:
We've been using `_sys_` for all of the platform dependent code so far, e.g. `golioth_sys_sem_create()`. I considered splitting out the more OS related code (semaphores, timers, threads, etc.) from the rest of it (right now just time, but would include these functions as well). That would also make it clearer for ports that use the same OS, like how ESP-IDF and Modus both use FreeRTOS. What do you think?

Reply:
Okay, let's keep `_sys_` in the name. Makes sense!